operator/main.go

```diff
@@ -194,7 +205,9 @@ func runOperator(cmd *cobra.Command) {
 			log.WithError(err).Fatal("Unable to start Azure allocator")
 		}
 
-		startSynchronizingCiliumNodes(nodeManager)
+		if err := startSynchronizingCiliumNodes(nodeManager); err != nil {
+			log.WithError(err).Fatal("Unable to start synchronizing Cilium nodes")
+		}
```

Comment:
The code in this switch statement is basically the same except for the cloud provider. We can eliminate the duplication by extracting the minimal necessary information from the switch statement and then proceeding outside the switch statement with the generic operation, like calling `Init()`, and the rest. 

For example, it seems like the only piece of info we need out of this is the specific allocator and provider.

```
ipamAllocatorAWS, providerBuiltin := allocatorProviders["aws"]
```
```
ipamAllocatorAzure, providerBuiltin := allocatorProviders["azure"] 
```

Reply:
The code has changed during the last rebase, everything is handled in the one case now,